examples/seismic/tutorials/09_viscoelastic.ipynb

```diff
@@ -0,0 +1,338 @@
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Source\n",
+    "src = RickerSource(name='src', grid=model.grid, f0=f0, time_range=time_range)\n",
+    "src.coordinates.data[:] = np.array([100., 50., 35.])\n",
+    "\n",
+    "# The source injection term\n",
+    "src_xx = src.inject(field=sxx.forward, expr=src*s)\n",
+    "src_yy = src.inject(field=syy.forward, expr=src*s)\n",
+    "src_zz = src.inject(field=szz.forward, expr=src*s)\n",
+    "\n",
+    "# fdelmodc reference implementation\n",
+    "eq_vx = Eq(vx.forward, damp*vx + damp*ro*s*(sxx.dx + sxy.dy + sxz.dz))\n",
+    "eq_vy = Eq(vy.forward, damp*vy + damp*ro*s*(sxy.dx + syy.dy + syz.dz))\n",
+    "eq_vz = Eq(vz.forward, damp*vz + damp*ro*s*(sxz.dx + syz.dy + szz.dz))\n",
+    "\n",
+    "eq_sxx = Eq(sxx.forward, damp*sxx \\\n",
+    "    + damp*s*pi*t_ep/t_s*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - damp*2.*s*mu*t_es/t_s*(vy.forward.dy+vz.forward.dz) \\\n",
+    "            + damp*s*rxx.forward)\n",
+    "eq_syy = Eq(syy.forward, damp*syy \\\n",
+    "    + damp*s*pi*t_ep/t_s*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - damp*2.*s*mu*t_es/t_s*(vx.forward.dx+vz.forward.dz) \\\n",
+    "            + damp*s*ryy.forward)\n",
+    "eq_szz = Eq(szz.forward, damp*szz +\\\n",
+    "    damp*s*pi*t_ep/t_s*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - damp*2.*s*mu*t_es/t_s*(vx.forward.dx+vy.forward.dy) \\\n",
+    "            + damp*s*rzz.forward)\n",
+    "eq_sxy = Eq(sxy.forward, damp*sxy \\\n",
+    "    + damp*s*mu*t_es/t_s*(vx.forward.dy+vy.forward.dx) + damp*s*rxy.forward)\n",
+    "eq_sxz = Eq(sxz.forward, damp*sxz \\\n",
+    "    + damp*s*mu*t_es/t_s*(vx.forward.dz+vz.forward.dx) + damp*s*rxz.forward)\n",
+    "eq_syz = Eq(syz.forward, damp*syz \\\n",
+    "    + damp*s*mu*t_es/t_s*(vy.forward.dz+vz.forward.dy) + damp*s*ryz.forward)\n",
+    "\n",
+    "eq_rxx = Eq(rxx.forward, damp*rxx \\\n",
+    "    - damp*s*1./t_s*(rxx+pi*(t_ep/t_s-1)*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - 2*mu*(t_es/t_s-1)*(vz.forward.dz+vy.forward.dy)))\n",
+    "eq_ryy = Eq(ryy.forward, damp*ryy \\\n",
+    "    - damp*s*1./t_s*(ryy+pi*(t_ep/t_s-1)*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - 2*mu*(t_es/t_s-1)*(vx.forward.dx+vy.forward.dy)))\n",
+    "eq_rzz = Eq(rzz.forward, damp*rzz \\\n",
+    "    - damp*s*1./t_s*(rzz+pi*(t_ep/t_s-1)*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - 2*mu*(t_es/t_s-1)*(vx.forward.dx+vy.forward.dy)))\n",
+    "eq_rxy = Eq(rxy.forward, damp*rxy \\\n",
+    "    - damp*s*1/t_s*(rxy+mu*(t_es/t_s-1)*(vx.forward.dy+vy.forward.dx)))\n",
+    "eq_rxz = Eq(rxz.forward, damp*rxz \\\n",
+    "    - damp*s*1/t_s*(rxz+mu*(t_es/t_s-1)*(vx.forward.dz+vz.forward.dx)))\n",
+    "eq_ryz = Eq(ryz.forward, ryz \\\n",
+    "    - damp*s*1/t_s*(ryz+mu*(t_es/t_s-1)*(vy.forward.dz+vz.forward.dy)))"
```

Comment:
Double check equations too there was a sign error in one of the elastic notebook.